### DISCUSSION TITLE
How to make a trace propagate further?

### DISCUSSION THREAD

**Discussion by rushsky518** [apache/skywalking#9546] [Q&A]
Hi everyone, i have a question about Skywalking trace. 
I marked a Netty Handler method with `@Trace` annotation, then it starts a span and finally ends at a biz method, but i expect the trace info continue to propagate, what should i do?

**Comment by pg-yang** [comment]
Where do you want to propagate ? 
If  you  mean  propagate  trace info  to  downstream , Our  plugin  should  cover   most library .
In a word, It  depends on  what library  you  used  for  requesting external service  , or  your  thread  model .

**Reply by rushsky518** [reply]
Thanks a lot, i noticed that TraceCrossThread can handle trace info in multi-thread scenario.  However we use projectrecator in the project. 
Maybe i have to write a plugin.

**Reply by pg-yang** [reply]
Yes . the  main reason is  thread-cross between  Handler and RPC framework that used for requesting downstream . 
If you write plugin , you could  reference https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/
